lib/absinthe/phase/schema/validation/default_enum_value_present.ex

```diff
@@ -58,6 +56,25 @@ defmodule Absinthe.Phase.Schema.Validation.DefaultEnumValuePresent do
     node
   end
 
+  defp value_conforms_to_enum(%Blueprint.TypeReference.List{of_type: of_type}, value, enum_values)
+       when is_list(value) do
+    value
+    |> Enum.map(&value_conforms_to_enum(of_type, &1, enum_values))
+    |> Enum.find({:ok, value}, &match?({:error, _}, &1))
+  end
+
+  defp value_conforms_to_enum(%_{of_type: of_type}, value, enum_values) do
+    value_conforms_to_enum(of_type, value, enum_values)
+  end
+
+  defp value_conforms_to_enum(_, value, enum_values) do
+    if value in enum_values do
+      {:ok, value}
+    else
+      {:error, value}
+    end
+  end
```

Comment:
Initially I started with a simpler version that would just return a boolean indicating if a value fits an enum. Such approach did however lead to quite misleading error messages, so I figured it's better to show which inner value actually failed to match the enum, and that's why this method returns it.